internal/generator/backend/template/representations/r1cs.go.tmpl

```diff
@@ -103,9 +102,9 @@ func (cs *R1CS) Solve(witness, a, b, c []fr.Element, opt backend.ProverConfig) (
 		if !check.Equal(&c[i]) {
 			if dID, ok := cs.MDebug[i]; ok {
 				debugInfoStr := solution.logValue(cs.DebugInfo[dID])
-				return solution.values, fmt.Errorf("%w: %s", ErrUnsatisfiedConstraint, debugInfoStr)
+				return solution.values, fmt.Errorf("%w: %s", errUnsatisfiedConstraint(i), debugInfoStr)
```

Comment:
As the error is not assertable anymore, then it doesn't make sense to wrap the error with "%w" anymore. Imo internally it returns a struct which contains the wrapped error and the message, but this overhead doesn't matter as the current function returns early.